src/hashmap.c

```diff
@@ -255,8 +255,8 @@ static Obj _DS_Hash_SetOrAccValue(Obj ht, Obj key, Obj val, Obj accufunc)
     }
     else {
         DS_IncrementCounterInPlist(ht, POS_USED, INTOBJ_INT(1));
-
-        SET_ELM_PLIST(keys, idx, key);
+        Obj immutableKey = CopyObj(key, 0);
```

Comment:
I'd very slightly favor `key = CopyObj(key, 0);` to avoid accidentally using the original key later on (in the spirit of defensive coding). Of course in this case, one can quite easily see that it's not the case, so it's OK to leave it.